src/tasks/batch_integration/methods/scanvi/config.vsh.yaml

```
@@ -0,0 +1,44 @@
+# The API specifies which type of component this is.
+# It contains specifications for:
+#   - The input/output files
+#   - Common parameters
+#   - A unit test
+__merge__: ../../api/comp_method_embedding.yaml
+
+functionality:
+  # A unique identifier for your component (required).
+  # Can contain only lowercase letters or underscores.
+  name: scanvi
+
+  # Metadata for your component
+  info:
+    # A relatively short label, used when rendering visualisarions (required)
+    label: Scanvi
+    # A one sentence summary of how this method works (required). Used when 
+    # rendering summary tables.
+    summary: "ScanVI is an extension of scVI but instead using a Bayesian semi-supervised approach for more principled cell annotation. Run as implemented by Luecken et al. 2022."
+    description : "TODO"
+    reference: "lopez2018deep"
+    repository_url: "https://github.com/YosefLab/scvi-tools"
+    documentation_url: "https://github.com/YosefLab/scvi-tools#readme"
+    v1:
+      path: openproblems/tasks/_batch_integration/batch_integration_graph/methods/scanvi.py
+      commit: 29803b95c88b4ec5921df2eec7111fd5d1a95daf
+    preferred_normalization: log_cpm
+    variants:
+      scanvi_full_unscaled:
+      scanvi_hvg_unscaled:
+        hvg: true
+  resources:
+    - type: python_script
+      path: script.py
+platforms:
+  - type: docker
+    image: ghcr.io/openproblems-bio/base_r:1.0.0
+    setup:
+      - type: python
+        pypi:
+          - scvi-tools
+          - scanpy
+          - scib==1.1.3
+  - type: nextflow
```

Add directives for nextflow: you can change the labels to e.g. `highmem` , `midmem` depending on the requirements of the method and how much memory or cpu it needs.

```suggestion
  - type: nextflow
    directives:
      label: [ lowmem, lowcpu ]
```

---

It was not in the docs but will be added and in the schemas